### DISCUSSION TITLE
Open Questions about syncs

### DISCUSSION THREAD

**Discussion by arthurgassmann** [open-feature/flagd#1283] [Q&A] [answered]
Hello, first of all thanks for your flagd contribution to open source!

We are evaluating on using flagd and have some open questions.

- If we want to add a new sync source at runtime, do we always need to restart flagD? If so, is it possible to have this dynamically?
- Is a partial gRPC sync of flags possible? Or is the complete set of flags required?
- Is it planned if there are syncs for db sources? such as nosql for example? Is there a roadmap for flagd?

Thanks a lot for your response, we would love to collaborate on that

**Comment by beeme1mr** [comment, marked as answer]
Hey @arthurgassmann, thanks for reaching out.

It's not possible to dynamically add new syncs at runtime. What use case did you have in mind?

The flag sync only supports sending the full flag configuration. We originally had support for sending deltas but we found that the added complexity wasn't worth the maintenance burden. Since flagd supports merging multiple sources, it's possible to scope your flags however you would like.

There are no plans currently to support a DB source. We've considered adding support for [s3-compatible storage providers](https://github.com/open-feature/flagd/issues/864) but we'll only implement this if it works cross-platform. That would be the challenge with adding DB support. There's no clear DB dialect that would be universally useful. I would recommend putting a simple API in front of the DB that supports either the [flagd definition over HTTP](https://flagd.dev/reference/flag-definitions/) or the [gRPC sync](https://flagd.dev/reference/specifications/protos/#flagdsyncv1syncproto).

You also may be interested in joining the `#openfeature-flag` Slack channel in the [CNCF workspace](https://cloud-native.slack.com/messages/CHY2THYUU). It's a nice way to connect with the dev team and end users.

**Reply by arthurgassmann** [reply]
Thanks a lot @beeme1mr for your answer

The use case for new syncs at runtime was to add new endpoints or new files on runtime for seperation of different feature flags for different applications / teams. 

We are now playing around with the gRPC sync and check within Slack. Thanks a lot for the quick reply
